pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -16,36 +16,82 @@ package cloudrun
 
 import (
 	"context"
+	"sync/atomic"
 
 	"go.uber.org/zap"
 
-	"github.com/pipe-cd/pipecd/pkg/config"
-	"github.com/pipe-cd/pipecd/pkg/model"
+	provider "github.com/pipe-cd/pipecd/pkg/app/piped/cloudprovider/cloudrun"
 )
 
-type applicationLister interface {
-	List() []*model.Application
-}
-
-type Store struct {
+type store struct {
+	apps   atomic.Value
 	logger *zap.Logger
+	client provider.Client
 }
 
-type Getter interface {
+func (s *store) run(ctx context.Context) error {
+	const maxLimit = 500
+	var cursor string
+	svc := make([]*provider.Service, 0, maxLimit)
+	for {
+		ops := &provider.ListOptions{
+			Limit:         maxLimit,
+			LabelSelector: provider.MakeManagedByPipedLabel(),
+			Cursor:        cursor,
+		}
+		// Cloud Run Admin API rate Limits.
+		// https://cloud.google.com/run/quotas#api
+		v, next, err := s.client.List(ctx, ops)
+		if err != nil {
+			s.logger.Error("failed to list cloudrun services", zap.Error(err))
+			return err
```

Comment:
nit, sorry many times, I mean, to avoid being log duplicated, how about:
```suggestion
			return fmt.Errorf("failed to list cloudrun services: %w", err)
```

Reply:
I’m very sorry to bother you.🙇 
That't nice. let me apply it